gtk4/src/entry_buffer.rs

```diff
@@ -125,4 +135,67 @@ impl<O: IsA<EntryBuffer>> EntryBufferExtManual for O {
             );
         }
     }
+
+    fn connect_deleted_text<F: Fn(&Self, u16, u16) + 'static>(&self, f: F) -> SignalHandlerId {
+        unsafe extern "C" fn deleted_text_trampoline<P, F: Fn(&P, u16, u16) + 'static>(
+            this: *mut ffi::GtkEntryBuffer,
+            position: libc::c_uint,
+            n_chars: libc::c_uint,
+            f: glib::ffi::gpointer,
+        ) where
+            P: IsA<EntryBuffer>,
+        {
+            let f: &F = &*(f as *const F);
+            f(
+                &EntryBuffer::from_glib_borrow(this).unsafe_cast_ref(),
+                to_u16!(position),
+                to_u16!(n_chars),
+            )
+        }
+        unsafe {
+            let f: Box_<F> = Box_::new(f);
+            connect_raw(
+                self.as_ptr() as *mut _,
+                b"deleted-text\0".as_ptr() as *const _,
+                Some(transmute::<_, unsafe extern "C" fn()>(
+                    deleted_text_trampoline::<Self, F> as *const (),
+                )),
+                Box_::into_raw(f),
+            )
+        }
+    }
+
+    fn connect_inserted_text<F: Fn(&Self, u16, &str, u16) + 'static>(
+        &self,
+        f: F,
+    ) -> SignalHandlerId {
+        unsafe extern "C" fn inserted_text_trampoline<P, F: Fn(&P, u16, &str, u16) + 'static>(
+            this: *mut ffi::GtkEntryBuffer,
+            position: libc::c_uint,
+            chars: *mut libc::c_char,
+            n_chars: libc::c_uint,
+            f: glib::ffi::gpointer,
+        ) where
+            P: IsA<EntryBuffer>,
+        {
+            let f: &F = &*(f as *const F);
+            f(
+                &EntryBuffer::from_glib_borrow(this).unsafe_cast_ref(),
+                to_u16!(position),
+                &glib::GString::from_glib_borrow(chars),
+                to_u16!(n_chars),
```

Comment:
Should `chars` be limited to `n_chars`?

Reply:
As this needs a bit of testing, I added an example and i will try to use some of those APIs and see how that goes